crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -503,6 +558,17 @@ impl TransactableStorage for RocksDb {
     }
 }
 
+/// The `None` means overflow, so there is not following prefix.
+fn next_prefix(mut prefix: Vec<u8>) -> Option<Vec<u8>> {
+    for byte in prefix.iter_mut().rev() {
+        if byte != &u8::MAX {
+            *byte = byte.checked_add(1).expect("We've just checked it above");
+            return Some(prefix)
+        }
```

Comment:
```suggestion
        if let Some(new_byte) = byte.checked_add(1) {
            *byte = new_byte
            return Some(new_byte)
        }
```

Reply:
Alternatively,

```rust
/// The `None` means overflow, so there is not following prefix.
fn next_prefix(mut prefix: Vec<u8>) -> Option<Vec<u8>> {
    prefix.iter_mut().rev().find_map(|p| {
        *p = p.checked_add(1)?;
        Some(())
    })?;
    Some(prefix)
}
```

But I think @MitchTurner's suggestion is easiest to read.

Reply:
Yeah. I'm not a fan of early `return`, but I also don't love doing mutation with a `map`. So I think I prefer the succinct approach.